loopchain/baseservice/node_subscriber.py

```diff
@@ -68,60 +87,68 @@ def __del__(self):
                 logging.debug(f"Closing websocket connection to {self._target_uri}...")
                 await websocket.close()
 
-    async def subscribe(self, block_height, event: Event):
-        self._exception = None
-        self._subscribe_event = event
-        await self.close()
+    async def run(self, event, block_height):
+        await self._prepare_connection(event)
+        await self._handshake(block_height)
+        await self.subscribe_loop()
 
+    async def _prepare_connection(self, event):
+        self._subscribe_event = event
```

Comment:
Any reason to setting member variable subscribe_event here?

Reply:
I popped it out from that function. Thank you.